Objects/unicodeobject.c

```diff
@@ -9728,9 +9728,14 @@ split(PyObject *self,
         default:
             Py_UNREACHABLE();
         }
+    }
 
     kind2 = PyUnicode_KIND(substring);
     len2 = PyUnicode_GET_LENGTH(substring);
+    if (maxcount < 0) {
+        // if len2 == 0, it will raise TypeError.
+        maxcount = len2 == 0 ? 0 : (len1 / len2) + 1;
```

Comment:
`(len1 / len2)+1` is correct upper bound. (e.g. `len("aaaaa".split('a')) == 6`).

But `(Py_SSIZE_T_MAX / 1)+1` would overflow. (e.g. `if (maxcount < 0) maxcount=Py_SSIZE_T_MAX;`)